### ISSUE TITLE
Allow trailing commas in multiple line declarations

### ISSUE THREAD

**Issue by p2kmgcl** [liferay/liferay-frontend-guidelines#120]
## What is your proposal?
Allow using trailing commas on multiple lines and enabling it by default on prettier.

## Why would adopting this proposal be beneficial?
As we are enforced to sort lot of things alphabetically (ex. object properties), refactoring this kind of code with trailing commas is simpler that not having them.

![image](https://user-images.githubusercontent.com/800645/74446257-e8adcb00-4e77-11ea-8376-729484c6ae5d.png)
_(left good order, right bad order)_

**Comment by wincent** [comment]
I'd like to do this for exactly the reason you suggest (reordering should be easy), but there are two reasons why we might not have done this historically. At least, I recall asking @jbalsas about this about a year ago and there were reasons back then...

One was because IE will choke on the trailing comma, and we didn't use to transform everything with Babel. But this is no longer true! Basically everything — except theme JS, and there is hardly any of that — goes through Babel now.

The other reason why it might not pass is if Brian doesn't like it, but I only have a vague recollection of that... maybe I am misremembering. In Java, [I believe trailing commas are allowed in arrays](https://hanuska.blogspot.com/2007/02/trailing-comma-in-arrays.html), but I don't know how often they come up, or if there are any other places they'd be permitted, or if we use them in liferay-portal.

In terms of options here, I suspect we're more likely to get trailing commas in all the places they're allowed in ES5 (eg. objects, arrays), but less likely to get them in the more recently supported positions (eg. function arguments lists).

In short, if @jbalsas thinks this change might fly, I'd be all for trying it.

**Comment by jbalsas** [comment]
Let's go for it

**Comment by wincent** [comment]
Closing this in favor of https://github.com/liferay/liferay-npm-tools/issues/380, where it will actually be implemented.
